scripts/priorities.py

```diff
@@ -8,14 +8,31 @@
 import numpy as np
 from Bio.SeqIO.FastaIO import SimpleFastaParser
 from Bio.Seq import Seq
-from Bio import AlignIO
+from Bio import AlignIO, SeqIO
 from scipy import sparse
 
 
+def compactify_sequences(sparse_matrix, sequence_names):
+    sequence_groups = defaultdict(list)
+    for s, snps in zip(sequence_names, sparse_matrix):
+        ind = snps.nonzero()
+        vals = np.array(snps[ind])
+        if len(ind[1]):
+            sequence_groups[tuple(zip(ind[1], vals[0]))].append(s)
+        else:
+            sequence_groups[tuple()].append(s)
+
+    return sequence_groups
+
 INITIALISATION_LENGTH = 1000000
 
+def sequence_to_int_array(s, fill_value=110):
+    seq = np.frombuffer(s.lower().encode('utf-8'), dtype=np.int8).copy()
```

Comment:
This failed for me & I don't have time to debug currently:
```
Traceback (most recent call last):
  File "scripts/priorities.py", line 142, in <module>
    ref = sequence_to_int_array(SeqIO.read(args.reference, 'genbank').seq)
  File "scripts/priorities.py", line 31, in sequence_to_int_array
    seq = np.frombuffer(s.lower().encode('utf-8'), dtype=np.int8).copy()
AttributeError: 'Seq' object has no attribute 'encode'
```

Reply:
this might be due to Biopython version. It works for me. I added an explicit string cast.

Reply:
yes, this was added recently: https://github.com/biopython/biopython/blame/master/Bio/Seq.py#L900